nengo_viz/swi.py

```diff
@@ -409,23 +410,20 @@ def start(cls, port=80, asynch=True, addr='', browser=False):
             # server.requests might be modified from other threads, so we need
             # a copy which we get in a thread-safe way be slicing it with [:].
             if asynch and server.requests is not None:
-                for _, sock in server.requests[:]:
+                for _, sock in server.requests:
                     try:
                         sock.shutdown(socket.SHUT_RDWR)
                         sock.close()
                     except socket.error:
                         pass
 
-                first = True
-                for thread, _ in server.requests[:]:
+                start = time.clock()
+                for thread, _ in server.requests:
                     if thread.is_alive():
-                        # giving the first thread more time to close
-                        # effectively gives all threads more time to close
-                        thread.join(0.2 if first else 0.01)
-                        first = False
+                        thread.join(max(0.01, 0.2 - (time.clock() - start)))
```

Comment:
Arg, I should get into a habit of reading the most recent docs and not always the 2.7. `time.perf_counter` would probably be the right function to use, but does not exist before Python 3.3 which would require even more code.
I agree, let's just do a constant amount.